lib/asserts.nix

```
@@ -2,78 +2,122 @@
 
 rec {
 
-  /* Throw if pred is false, else return pred.
-     Intended to be used to augment asserts with helpful error messages.
+  /**
+    Throw if pred is false, else return pred.
+    Intended to be used to augment asserts with helpful error messages.
 
-     Example:
-       assertMsg false "nope"
-       stderr> error: nope
+    # Example
 
-       assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
-       stderr> error: foo is not bar, silly
+    ```nix
+    assertMsg false "nope"
+    stderr> error: nope
+    assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
+    stderr> error: foo is not bar, silly
+    ```
+
+    # Type
+
+    ```
+    assertMsg :: Bool -> String -> Bool
+    ```
+
+    # Arguments
```

```suggestion
    # Inputs
```

I've been using "Inputs" for the other functions outside `lib`, so to keep things consistent I'm suggesting we keep using that. Not a strong opinion, so if you provide any good point to use "Arguments" instead of "Inputs", I'll very likely agree on the change :)
(applies to other functions too)